OpenIDConnectClient.php5

```diff
@@ -403,28 +403,14 @@ class OpenIDConnectClient
       */
      private function get_key_for_header($keys, $header) {
          foreach ($keys as $key) {
-             if ($key->alg == $header->alg && $key->kid == $header->kid) {
+             if ((isset($header->kid) && $key->alg == $header->alg && $key->kid == $header->kid) 
+                  || ($key->kty == 'RSA')) {
                  return $key;
              }
          }
          throw new OpenIDConnectClientException('Unable to find a key for (algorithm, kid):' . $header->alg . ', ' . $header->kid . ')');
```

Comment:
This patch works great for me, but it needs one final enhancement. Please change line 411 from:

         throw new OpenIDConnectClientException('Unable to find a key for (algorithm, kid):' . $header->alg . ', ' . $header->kid . ')');

to

         if (isset($header->kid)) {
             throw new OpenIDConnectClientException('Unable to find a key for (algorithm, kid):' . $header->alg . ', ' . $header->kid . ')');
         } else {
             throw new OpenIDConnectClientException('Unable to find a key for RSA');
         }

Thanks!